src/api/registerApplicationResourceResolver.ts

```diff
@@ -17,6 +19,10 @@ export function registerApplicationResourceResolver(id: string, resolver: AppRes
     applicationResourceResolvers[id] = resolver;
     ext.emitters.onDidRegisterResolver.fire(resolver);
 
+    const compat = new CompatibleBranchDataProvider(resolver, 'azureResourceGroups.loadMore' /** TODO: what is the correct value for this? */);
+
+    ext.v2.api.registerApplicationResourceBranchDataProvider(id, compat as unknown as BranchDataProvider<ApplicationResource, ResourceModelBase>);
```

Comment:
What makes this `unknown` cast necessary? `CompatibleBranchDataProvider` does implement `BranchDataProvider`.

Reply:
It's because we have `getParent` defined in `CompatibleBranchDataProvider`, but in `BranchDataProvider` we have `getParent?: never;`. We need to remove one of those.

```
Argument of type 'CompatibleBranchDataProvider<ApplicationResource, AzExtTreeItem & ResourceModelBase>' is not assignable to parameter of type 'BranchDataProvider<ApplicationResource, AzExtTreeItem>'.
  Types of property 'getParent' are incompatible.
    Type '(_treeItem: AzExtTreeItem & ResourceModelBase) => Promise<AzExtTreeItem & ResourceModelBase>' is not assignable to type 'undefined'.
```

Reply:
Ah, I see. I would have thought that the `//@ts-expect-error` above the `getParent()` definition in `CompatibleBranchDataProvider` would work. Oh well!